lib/cmd/cmd_common.ml

```diff
@@ -31,23 +33,61 @@ let result_guard def_fn =
       | Error exn ->
         raise exn)
 
-let simply_print_response ~exn format =
-  (Async_shell.sh_one format)
-  >>= function
-  | Some result ->
-    print_string result;
-    return @@ Ok ()
-  | None ->
-    return @@ Error exn
+let get_cwd
+  : String.t Option.t -> String.t Deferred.t =
+  function
+  | Some cwd -> return cwd
+  | None -> Sys.getcwd ()
+
+let sh
+  : ?working_dir:String.t -> ?shell:String.t List.t -> String.t ->
+    (String.t, Exn.t) Deferred.Result.t =
+  fun ?working_dir ?(shell = ["bash"; "-c"]) cmd ->
+    get_cwd working_dir
+    >>= fun cwd ->
+    let actual_args = List.concat [shell; [cmd]] in
+    Process.run ~working_dir:cwd ~prog:"/usr/bin/env" ~args:actual_args ()
```

Comment:
But you can assume it accessible in the path. This is how I do it for nixos: https://github.com/samuelrivas/assorted-scripts/blob/master/nix/post-fixup.sh#L28

That means that the script/program would work normally (i.e. searching through the PATH) when run as is, but will use the exact dependency if installed through nixos

Reference:
https://nixos.org/wiki/Nix_Runtime_Environment_Wrapper